File: backend/models/Video.go
package models

import "gorm.io/gorm"

type Video struct {
	gorm.Model
	Title          string `gorm:"not null;default:'unknown'"`
	IsDefault      bool   `gorm:"default:false"`
	Duration       uint
	BitRate        uint
	BitDepth       uint
	Codec          string `gorm:"not null;default:'unknown'"`
	ColorSpace     string `gorm:"not null;default:'unknown'"`
	ColorPrimaries string `gorm:"not null;default:'unknown'"`
	Width          uint
	Height         uint
	AvgFrameRate   float32
	FrameRate      float32
	AspectRatio    string `gorm:"not null;default:'unknown'"`
	MovieID        uint
	Movie          Movie
}

File: web/src/types/Video.ts
export type Video = {
  ID: number;
  title: string;
  index: number;
  profile: string;
  aspectRatio: string;
  bitDepth: string;
  codec: "aac" | "mp3" | "copy";
  width: number;
  height: number;
  codedHeight: number;
  codedWidth: number;
  colorTransfer: string;
  colorPrimaries: string;
  colorSpace: string;
  frameRate: string;
  avgFrameRate: string;
  CreatedAt?: string | Date;
  UpdatedAt?: string | Date;
  DeletedAt?: string | Date | null;
};

export type VideoError = {
  error: {
    errorString?: string;
    errorException?: string;
  };
};
